src/ctapipe/image/extractor.py

```diff
@@ -1295,6 +1296,20 @@ def __call__(
         )
 
 
+class VarianceExtractor(ImageExtractor):
+
+    """
+    Extractor that calculates
+    the variance of each waveform
+    """
+
+    def __call__(
+        self, waveforms, tel_id
+    ) -> DL1CameraContainer:
+        variance = np.nanvar(waveforms,axis=2)
```

Comment:
Do we really just want to compute the variance of the hole waveform?

I know pedvar as the the variance over an integration window. 

You place as many integration windows as you can in the waveform and then compute the variance over the integrals.

Reply:
that might be interesting, however, I didn't test it... A simple variance like here works for sure. 
@ctoennis I think the idea of Max is nice and we can implement it alongside and compare later

Reply:
Just to be clear, that's not my idea but comes from the original ped var paper, I think by VERITAS, let me dig it up

Reply:
I will see how well this works compared to just looking at the variance directly.

Reply:
since the amplitude of the signal in bin N+1 does not depend on the amplitude of the signal in bin N, then the variance of a sample and the variance of charge are equivalent (cov(Ai, Aj) == 0). The only differences are statistical error on the calculated variance and the normalization.

Reply:
I'm not sure I understand the context, so apologies if I misunderstood what you've written - just wanted to say that if by "bin" you mean "time slice" for a given channel: consecutive samples are _not_ independent because the single-p.e. pulse shapes are several nanoseconds long (tens of nanoseconds for some cameras). Also electronics noise is usually not white.

Reply:
You understand it correctly and I was thinking of what you wrote. However we consider NSB radiation, which is commonly modeled as a Poisson process, i.e. the numbers of photons hitting the pixel in two consecutive time intervals are independent. Now, for the PMT-equipped camera the conversion of photons to p.e. will also be uncorrelated. This is not correct for SiPM cameras, where generalized Poisson shall be applied.